java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/typesolvers/JavaParserTypeSolver.java

```diff
@@ -41,12 +41,15 @@
 
     private TypeSolver parent;
 
-    private Map<String, CompilationUnit> parsedFiles = new HashMap<String, CompilationUnit>();
+    private Map<String, Optional<CompilationUnit>> parsedFiles = new HashMap<>();
     private Map<String, List<CompilationUnit>> parsedDirectories = new HashMap<>();
-    private Map<String, ReferenceTypeDeclaration> foundTypes=new HashMap<>();
+    private Map<String, ReferenceTypeDeclaration> foundTypes = new HashMap<>();
+    
+    private boolean hasValidSrcDir = false;
 
     public JavaParserTypeSolver(File srcDir) {
         this.srcDir = srcDir;
+        hasValidSrcDir = srcDir.exists() && srcDir.isDirectory();
```

Comment:
My thoughts is that I would demand an SSD :D
Seriously, maybe it makes sense to throw an Exception in the constructor then, because the directory is actually useless